stumpy/stump.py

```diff
@@ -190,6 +190,25 @@ def _compute_diagonal(
                 if T_B_subseq_isconstant[i + k] and T_A_subseq_isconstant[i]:
                     pearson = 1.0
 
+                # due to imprecision, the value of pearson may be outside
+                # of interval [-1, 1]
+                if pearson >= 1.0:
```

Comment:
I think it's simpler/cleaner to do:

```
pearson = np.clip(pearson, -1.0, 1.0)
```

Reply:
Actually, I think this would break backwards compatibility as np.clip is not supported in older versions of numba. Please ignore this comment (unless we are already using np.clip elsewhere in our code)

Reply:
Actually, our minimum `numba` version in `requirments.txt` has `numba>=0.54` and this is the version that includes `np.clip` so I think we are fine using `np.clip`!

Reply:
Thanks for letting me know! I will try it then.